zbsprimarystorage/zbsprimarystorage/zbsagent.py

```
@@ -215,13 +215,22 @@ def __init__(self):
 
     @replyerror
     def ping(self, req):
-        rsp = AgentResponse()
+        rsp = PingRsp()
+
+        def get_mds_external_addr(mds_map):
+            for res in mds_map.result:
+                externalAddr = res.externalAddr.split(':')[0]
+                cmd = 'ip route | grep -w "proto kernel" | grep -w %s > /dev/null' % externalAddr
+                if bash_r(cmd) == 0:
```

Ruff (0.11.9)

`bash_r` may be undefined, or defined from star imports

(F405)